docs/tcpAgent.md

```diff
@@ -0,0 +1,140 @@
+# Launching inbound TCP agents
+Jenkins provides a number of ways of connecting remote agents.
+Two of the most popular are SSH agents and inbound TCP agents.
+SSH agents, most commonly used on Unix platforms, are master initiated.
+The master creates the connection when it needs.
+Inbound TCP agents, most commonly used on Windows platforms, are client initiated.
+The client must first connect to the master and then the master sends commands as needed.
+These were formerly known as JNLP agents, but that name was erroneous and confusing.
+This document describes some of the primary mechanisms for launching inbound TCP agents.
+
+For additional information about Jenkins agents see [Distributed builds](https://wiki.jenkins.io/display/JENKINS/Distributed+builds#Distributedbuilds-HavemasterlaunchagentonWindows).
+
+## Launch mechanisms
+Part of the agent status page for TCP agents looks something like this:
+![Tcp agent status UI](tcpAgentStatus.jpg)
+This shows a couple of launch methods, but there are other mechanisms.
+If you use a different mechanism than clicking the "Launch" button, this page contains parameters that you will need. 
+The long string of hex digits is a secret key that the client needs to establish the connection. 
```

Comment:
Reading "secret key" here made me do some digging. While there is indeed a secret key (one generated from HmacSHA256 currently), what is actually returned and used here is a MAC of the name of the agent computer. As a budding crypto nerd, I'd be happy if you could make this sentence less ambiguous somehow.

Reply:
I'm open to suggestions on how to re-word it. I don't think this is the place to go into details of the construction of the key, though. That could be useful in a more detailed, technical implementation document. This document is about giving users information on how to launch the agent.

Reply:
It almost makes sense to call it a "secret token" rather than a "secret key", but then it seems as though the common use of the word "token" is for something revocable (like the crumbs and API keys sorta). Maybe you can simplify it to just "secret" rather than "secret key"?

Reply:
Come to think of it, each secret is agent-specific, so maybe you can call it an "agent secret"? I'm almost tempted to say "agent identity", but it works completely different from the instance identity thing.

Reply:
I'm not as fond of using "token". It seems to commonly have a different meaning, as you describe.

"secret" might work, but though technically correct I feel it might be confusing to a number of users. It's called "secret key" a couple of places and I think that makes more sense to general users. I don't think adding the modifier "agent" onto the beginning helps the user much.

I'm not committed to the current term, but I'm not convinced the alternatives help much.

Reply:
Well, as long as we aren't actually using any AES or other symmetric keys in an ambiguous context, then keeping "secret key" would be fine with me.